### PR TITLE
Integration test for R2 rules without default pipeline

### PR DESCRIPTION
**Pull request by shaan420**
R2 supports mapping rules and rollup rules that can be associated with an incoming metrics. Mapping rules are used to override the default storage policy (resolution and retention). Rollup rules are used to create a new aggregated rollup metric from original metric. Both types of rules are modeled as a `Pipeline` struct nested inside the `StagedMetadata` returned by matcher when a metric is matched to a either type of the rules.

The ForwardMatch() result is a bit more complex: It has,
1. a metadata slice for the existing metric which includes any mapping rules AND multilevel rollup rules where a rollup is not the first level aggregation
2. another metadata slice for rollup metrics which includes pipelines with roll up as the first level aggregation.
Another nuance is that the various targets in a rollup rule all become pipelines within the same stagedMetadata struct. We have multiple stagedMetadatas associated with the same metric when we forward match on a time range and multiple versions of the rule are matched.

For multi-level R2 pipelines containing a rollup operation, the first level aggregation must be performed on the original metric. When keepOriginal is configured to be false, we still need the original metric to compute the transform/aggregation and subsequently apply the rollup. However we should not emit the original metric itself. This emission occurs because of the default pipeline in the R2 rule. 
The integration test validates the behavior of sending in a metric without a default pipeline but containing a multilevel R2 pipeline to make sure that the original metric is not emitted whereas the rolled up metric is. 

**What this PR does / why we need it**:

Fixes #
The PR does not fix anything nor does it add any new feature. It simply validates the behavior of metric processing when the incoming staged metadata does not contain a default R2 pipeline.

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

**Comment by abliqo** [comment]
Overall look good, with nits. Please update the description to provide more context about the caveat with matcher. Otherwise, it's not obvious for less familiar folks.

**Review by abliqo — APPROVED** [review]
LGTM with naming nits. Please update the test name to make its meaning more clear.

File: src/aggregator/integration/one_client_keep_original_false_test.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+	require.NoError(t, testServer.stopServer())
+	log.Info("server is now down")
+
+	actual := testServer.sortedResults()
+	for _, m := range actual {
+		log.Info("actual metric", zap.String("id", m.ID.String()))
+	}
+
+	for _, m := range metricsWithExpectations {
+		if m.expectedInOutput {
```
[the remark this answers is not in the document]
Comment (shaan420) [reply src/aggregator/integration/one_client_keep_original_false_test.go]: Done

File: src/aggregator/integration/integration_data.go [hunk 2]
```diff
@@ -125,6 +128,57 @@ var (
...
+			Tombstoned:   false,
+			Metadata: metadata.Metadata{
+				Pipelines: []metadata.PipelineMetadata{
+					testDefaultPipelineMetadata,
+					testCustomPipelineMetadata,
+				},
+			},
+		},
+	}
+	testCustomStagedMetadatasWithoutDefaultPipeline = metadata.StagedMetadatas{
```
[the remark this answers is not in the document]
Comment (shaan420) [reply src/aggregator/integration/integration_data.go:171]: Done

File: src/aggregator/integration/integration_data.go [hunk 3]
```diff
@@ -125,6 +128,57 @@ var (
...
+			},
+		},
+	}
+	testDefaultPipelineMetadata = metadata.PipelineMetadata{
+		AggregationID: maggregation.DefaultID,
+		StoragePolicies: policy.StoragePolicies{
+			policy.NewStoragePolicy(time.Second, xtime.Second, time.Hour),
+		},
+	}
+	testCustomStagedMetadatasWithDefaultPipeline = metadata.StagedMetadatas{
```
[the remark this answers is not in the document]
Comment (shaan420) [reply src/aggregator/integration/integration_data.go:159]: Done

File: src/aggregator/integration/one_client_keep_original_false_test.go [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+// whereas foo2 has a default pipeline.
+// The output of the test should therefore be 2 metrics in total:
+// foo2 and bar1. foo1 should not be present in the output as it
+// does not have a default pipeline.
+// Context: When keepOriginal is false, we remove the default
+// pipeline from the stagedMetadata of the original metric.
+// This makes the aggregator drop the original metric but
+// continues to aggregate and emit the rolled up metric which
+// is what we expect.
+func TestOneClientKeepOriginalFalse(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (shaan420) [reply src/aggregator/integration/one_client_keep_original_false_test.go:52]: Done

File: src/aggregator/integration/one_client_keep_original_false_test.go [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+	log.Info("server is now the leader")
+
+	client := testServer.newClient(t)
+	require.NoError(t, client.connect())
+
+	ts := time.Now().Truncate(time.Hour)
+
+	clock.SetNow(ts)
+
+	metricsWithExpectations := []struct {
```
[the remark this answers is not in the document]
Comment (shaan420) [reply src/aggregator/integration/one_client_keep_original_false_test.go]: Done

**Pull request merged**